---
title: Metabolic Network Analysis and Metatranscriptomics of a Cosmopolitan and Streamlined Freshwater Lineage
author: Joshua J. Hamilton^1\*^, Sarahi L. Garcia^2, Brittany S. Brown^1, Ben O. Oyserman^3, Francisco Moya-Flores^3, Stefan Bertilsson^2, Rex R. Malmstrom^4, Katrina T. Forest^1, Katherine D. McMahon^1,3
abstract: \* Correspondence&#58; Joshua J. Hamilton, jjhamilton2@wisc.edu
date: ^1^ Department of Bacteriology, University of Wisconsin-Madison, Madison, WI, USA; ^2^ Department of Ecology and Genetics, Limnology and Science for Life Laboratory, Uppsala University, Uppsala, Sweden; ^3^ Department of Civil and Environmental Engineering, University of Wisconsin-Madison, Madison, WI, USA; ^4^ United States Department of Energy Joint Genome Institute, Walnut Creek, CA, USA
---

# Abstract

An explosion in the number of available genome sequences obtained through metagenomics and single-cell genomics has enabled a new view of the diversity of microbial life, yet we know surprisingly little about how microbes interact with each other or their environment. In fact, the majority of microbial species remain uncultivated, with many insights about an organism's ecological niche arising from metabolic reconstruction of its genome content. In this work, we demonstrate how the "seed set framework" enables high-throughput, computational analysis of metabolic reconstructions, while providing new insights into a microbe's metabolic capabilities, such as nutrient sources and essential metabolites. We apply this framework to members of the ubiquitous freshwater Actinobacterial lineage acI, confirming and extending previous experimental and genomic observations that suggest acI bacteria exhibit a heterotrophic lifestyle reliant on peptides and saccharides. We also present the first metatranscriptomic study of the acI lineage. These results reveal strong expression of transport proteins and the light-harvesting protein actinorhodopsin, suggesting the acI are capable of photoheterotrophy.

# Introduction

Microbial communities support essential ecosystem functions, ranging from nutrient cycling in the environment to influencing human health and disease [@Falkowski2008a; @Blaser2016]. However, the majority of microbial species remain uncultivated, which has posed a significant challenge to understanding their physiology and metabolism. Recent advances in sequencing technology and bioinformatics have made available reference genomes for community members from diverse environments [@Sangwan2016] that can be to infer links between individual microbe's genome content to its metabolic traits, a concept referred to as "reverse ecology" [@Levy2012].

Metabolic reconstructions represent a common entry point to reverse ecological analyses, in which an organism's metabolic capabilities are inferred from its genome content using pathway databases such as KEGG, MetaCyc, or SEED [@Kanehisa2017; @Caspi2016; @Overbeek2014]. Other approaches take cues from systems biology, focusing not just on the "parts-list" of an organism's genome content, but the ways in which those parts come together and interact. These approaches rely on metabolic network reconstructions [@Feist2009a; @Thiele2010c], structured summaries of an organism's metabolic capabilities as defined by its enzymes and their associated biochemical reactions (its metabolic network). These reconstructions can then be analyzed using metabolic network graphs, mathematical objects in which biochemical reactions are represented as connections between substrates and products [@Levy2012]. One such graph-based, reverse ecology approach is the _seed set framework_, that computes an organism's _seed set_, the set of compounds that the organism cannot synthesize on its own and must exogenously acquire from its environment [@Borenstein2008]. As such, these compounds may represent both _auxotrophies_, essential metabolites for which biosynthetic routes are missing, and _nutrients_, for which degradation (not synthesis) routes are present in the genome. The seed set framework offers potential advantages over other reconstruction-based approaches, as 1) metabolic network graphs can be rapidly analyzed computationally, 2) a network-centric approach makes no _a priori_ assumptions about which metabolic pathways may be important for an organism's niche, and 3) identification of seed compounds facilitates a focused analysis by identifying those compounds that an organism must obtain from its environment.

Freshwater lakes are ideal systems in which to apply the seed set framework, as long-term monitoring has revealed the ecology of dominant bacterial lineages [@Newton2011a], and reference genomes for these lineages are now readily available [@Martinez-Garcia2012; @Garcia2013; @Ghai2014; @Ghylin2014; @Tsementzi2014; @Garcia2015; @Bendall2016]. Of the freshwater bacteria, uncultivated Actinobacteria of the acI lineage are among the most abundant [@Zwart1998; @Glockner2000; @Zwart2002]. The acI have been phylogenetically divided into three clades (acI-A, acI-B, and acI-C) and thirteen tribes on the basis of their 16S rRNA gene sequences [@Newton2011a], and the abundance of these free-living ultramicrobacteria suggests they play a role in nutrient cycling in diverse freshwater systems [@Glockner2000; @Newton2006; @Wu2006b; @Newton2007; @Wu2007; @DeWever2008; @Humbert2009; @Ghai2012].

To identify the nutrient transformations these bacteria may mediate, the metabolism of the acI lineage has been extensively studied in a community context using both DNA sequencing and single-cell targeted experiments. Studies using fluorescent _in situ_ hybridization (FISH) and catalyzed reporter deposition (CARD) or microautoradiography (MAR) reveal that the acI are capable of consuming amino acids [@Salcher2010; @Salcher2013], glucose [@Buck2009; @Salcher2013], N-acetylglucosamine (NAG) [@Beier2011; @Eckert2012; @Eckert2013], the deoxynucleoside thymidine [@Perez2010; @Salcher2013], and acetate [@Buck2009]. However, due to methodological limitations of some FISH probes, the phylogenetic resolution of these studies is coarse compared to the phylogenetic resolution made possible by 16S rRNA gene sequencing.

To overcome this limitation, metabolic reconstructions of single-cell genomes (SAGs) and metagenome-assembled genomes (MAGs) have been used to propose substrate uptake capabilities for members of clades acI-A and acI-B. These studies indicate members of both clades are capable of consuming a wide array of N-containing compounds, including ammonium, branched-chain amino acids, polyamines, di-peptides, and cyanophycin [@Ghylin2014; @Garcia2015], with members of clade acI-A also capable of consuming oligopeptides [@Ghylin2014]. Members of these two clades are also capable of consuming numerous saccharides, including xylose, ribose, and arabinose [@Garcia2013; @Ghylin2014; @Garcia2015] as well as poly- and oligo-saccharides [@Ghylin2014; @Garcia2015]. Members of clade acI-B are also predicted to consume sucrose and maltose [@Garcia2015]. Finally, a recent study of a metagenome-assembled genome from clade acI-B predicted that some members of the clade are unable to synthesize a number of essential vitamins and amino acids [@Garcia2015]. In the aggregate, these results indicate that acI are photoheterotrophs, making a living on a diverse array of N-rich compounds, sugars, and oligo- and poly-saccharides. The acI lineage does not appear to be metabolically self-sufficient, relying on other organisms for the production of essential nutrients.

In this work, we develop a computational pipeline to automate the calculation of an organism's substrate utilization capabilities using the seed set framework, thereby facilitating high-throughput analysis of genomic data. We expand existing analyses of the acI lineage by applying the seed set framework to a reference genome collection of 36 freshwater acI genomes covering all three acI clades, including for the first time genomes from clade acI-C. To do so, we developed a Python package to predict seed compounds, using the seed set framework and metabolic network reconstructions generated from KBase (http://kbase.us). The seed compounds predicted by our analysis are in agreement with previous experimental and genomic observations, confirming the ability of our method to predict an organism's auxotrophies and nutrient sources. To validate and complement these predictions, we conducted the first metatranscriptomic analysis of gene expression in the acI lineage. Knowledge of seed compounds enhanced interpretation of the metatranscriptome results by facilitating a focused analysis. Additional analysis shows that the acI express a diverse array of transporters that we hypothesize may contribute to their observed dominance in a wide variety of aquatic systems.

# Materials and Methods

## A Freshwater Reference Genome Collection

This study relies on an extensive collection of freshwater bacterial genomes, containing MAGs obtained from two metagenomic time-series from two Wisconsin lakes [@Bendall2016; @Garcia2016a], as well as SAGs from three lakes in the United States [@Martinez-Garcia2012]. Additional information about this genome collection can be found in the Supplemental Online Material.

## Metatranscriptome Sampling and Sequencing

This study used four metatranscriptomes obtained as part of a 24-hour sampling experiment designed to identify diel trends in freshwater microbial communities. Samples were collected from the top of the water column (depth <1m) from Lake Mendota (Madison, WI, USA) on August 20 and 21, 2015. For each sample, between 200 and 400 mL lake water was filtered onto a 0.2 μm polyethersulfone filter (Supor, Pall Corp), flash frozen in liquid nitrogen, and stored at -80°C until extraction.

Samples were subject to TRIzol-based RNA extraction (Thermo Fisher Scientific, Waltham, MA) followed by phenol-chloroform separation and RNA precipitation. RNA was purified following an on-column DNAse digestion using the RNase-Free DNase Set (Qiagen, Venlo, Netherlands) and cleaned up with the RNeasy Mini Kit (Qiagen, Venlo, Netherlands). RNA was then sent to the University of Wisconsin-Madison Biotechnology Center (https://www.biotech.wisc.edu) for sequencing. There, samples were prepared for sequencing using the TruSeq RNA Library Prep Kit v2 (Illumina, San Diego, CA), with a ribosomal RNA (rRNA) depletion step using the Ribo-Zero rRNA Removal Kit (Bacteria) (Illumina). The resulting cDNA libraries were pooled in an equimolar ratio, and sequenced on an Illumina HiSeq2500 platform.

Raw paired-end reads were then trimmed using Sickle [@Joshi2011] and merged using FLASH [@Magoc2011]. Sickle was run using default parameters, and FLASH was run with a maximum overlap of 100 nucleotides (`M = 100`). Finally, additional rRNA and ncRNA sequences were removed using SortMeRNA [@Kopylova2012] using default parameters. SortMeRNA was run using eight built-in databases for bacterial, archaeal, and eukaryotic small and large ribosomal subunits and ncRNAs, derived from the SILVA 119 [@Quast2013] and RFAM [@Nawrocki2015] databases.

Metadata about the four samples used in this study can be found in Table S1, and the raw RNA sequences can be found on the National Center for Biotechnology Information (NCBI) website under BioProject PRJNA362825. Additional information, including all protocols and scripts for sample collection, RNA extraction, sequencing, and bioinformatic analysis can be found on Github (https://github.com/McMahonLab/OMD-TOILv2, DOI:######).

## Identification and Taxonomic Assignment of acI SAGs and MAGs

Novel acI SAGs were identified and classified to the tribe level using partial 16S rRNA genes and a reference taxonomy for freshwater bacteria, as described in the Supplemental Online Material. Actinobacterial MAGs were identified using taxonomic assignments from a subset of conserved marker genes, as described previously [@Bendall2016]. Phylogenetic analysis of acI SAGs and Actinobacterial MAGs was performed using a concatenated alignment of single-copy marker genes obtained via Phylosift [@Darling2014]. Maximum likelihood trees were generated using RAxML [@Stamatakis2014] using the automatic protein model assignment option (PROTGAMMAAUTO) and 100 bootstraps.

Previous phylogenetic analysis using 16S rRNA gene sequences indicates the acI lineage is monophyletic with three distinct monophyletic clades [@Newton2011a]. In the phylogenetic tree built from concatenated marker genes, the acI SAGs fell within a cluster containing the same topology as the 16S tree. MAGs falling within this cluster were classified as acI, enabling MAGs to be classified as clade acI-A, acI-B, or acI-C based on the location of SAGs within the tree. acI MAGs were then classified as belonging to a particular tribe if they formed a monophyletic group with a SAG from that tribe.

## Genome Completeness

CheckM [@Parks2015] was used to estimate genome completeness based on 204 single-copy marker genes conserved across the phylum Actinobacteria.

## Metabolic Network Reconstruction and Reverse Ecology

In the seed set framework, an organism's metabolism is represented via a metabolic network graph, in which nodes denote compounds and edges denote enzymatically-encoded biochemical reactions linking substrates and products [@Jeong2000]. Allowable biochemical transformations can be identified by drawing paths along the network, in which a sequence of edges connects a sequence of distinct vertices. Formally, the seed set of the network is defined as the minimal set of compounds that cannot be synthesized from other compounds in the network, and whose presence enables the synthesis of all other compounds in the network [@Borenstein2008]. Informally, seed compounds may represent both auxotrophies, essential compounds that the organism cannot synthesize, and nutrients, compounds from which other metabolites can be synthesized. In our implementation of the seed set framework, metabolic network graphs were generated as follows.

## Genome Annotation and Reconstruction Processing

Genome annotations were performed and metabolic network reconstructions were built using KBase (http://kbase.us/). Contigs for each genome were uploaded to KBase and annotated using the "Annotate Microbial Contigs" method with default options, which uses components of the RAST toolkit [@Brettin2015; @Overbeek2014] for genome annotation. Metabolic network reconstructions were obtained using the "Build Metabolic Model" app with default parameters, which relies on the Model SEED framework [@Henry2010a] to build a draft reconstruction.

Reconstructions were then pruned and converted to metabolic network graphs. During this process, exchange and transport reactions were removed from the reconstruction, to prevent extracellular metabolites from being identified as seed compounds. The biomass reaction was also removed, as KBase generates generalized biomass equations that may not reflect acI-specific biomass requirements. Finally, DNA/RNA replication reactions were removed, as these reactions do not represent metabolic processes. Reactions in the reconstructions were then mass- and charge-balanced. Next, currency metabolites (compounds used to carry electrons and functional groups) and highly-connected compounds (those that participate in many reactions, such as CO2 and O2) were removed to ensure paths in the resulting metabolic network graph would be biologically meaningful [@Ma2003] (see Figure S1 for an example). Finally, the metabolic network graph was extracted from the reconstruction, to enable graph-theoretical identification of the network's seed set. An illustration of this process can be found in Figure S1 in the Supplemental Online Material.

Many of the individual acI genomes are incomplete (see Results). Therefore, composite metabolic network graphs were constructed for each clade, to increase the accuracy of seed identification. To do so, all genome-level metabolic network graphs for all genomes within each acI clade were combined to generate a composite clade-level metabolic network graph. Beginning with two genomes, nodes and edges unique to the second genome are identified and appended to the network graph for the first genome, giving a composite metabolic network graph. The process is repeated for each genome, until all of the network graphs have been incorporated into the composite. An illustration of this process and additional details can be found in Figure S2 and in the Supplemental Online Material.

All steps were implemented using custom Python scripts, freely available as part of the reverseEcology Python package (https://pypi.python.org/pypi/reverseEcology/, DOI:######).

## Computation and Evaluation of Seed Compounds

Seed compounds for each composite clade-level metabolic network graph were calculated using the seed set framework [@Borenstein2008]. Briefly, the graph is decomposed into its strongly connected components (SCCs), sets of nodes such that each node in the set is reachable from every other node. Seed compounds can then be found by identifying source components (components with no incoming edges) on the condensation of the original graph, a representation in which each SCC is represented as a single vertex. Here, each source component represents a seed set, and the nodes within that vertex represent seed compounds. If a seed set contains multiple seed compounds, each compound is assigned a weight equal to the inverse of the number of compounds in the seed set. An illustration of this process can be found in Figure S3 in the Supplemental Online Material. Finally, all predicted seed compounds were manually evaluated to identify those that may be biologically meaningful. Examples are given in the Supplemental Online Material.

## Identification of Transported Compounds

Microbes may be capable of transporting compounds that are not strictly required for growth, and comparing such compounds to predicted seed compounds can provide additional information about an organism's ecology. Thus, we used the metabolic network reconstructions for the acI genomes to systematically characterize the transport capabilities of the acI lineage. For each genome, we identified all transport reactions present in its metabolic network reconstruction. Gene-protein-reaction associations (GPRs) for these reactions were manually curated to remove unannotated proteins, group genes into operons (if applicable), and to identify missing subunits for multi-subunit transporters. These genes were then mapped to their corresponding COGs, and GPRs were grouped on the basis of their mapped COGs. Finally, consensus annotations within each clade were used to identify likely substrates for each of these groups.

## Integrating Reverse Ecology with Metatranscriptomics

## Protein Clustering, Metatranscriptomic Mapping, and Clade-Level Gene Expression

OrthoMCL [@Li2003] was used to identify clusters of orthologous groups (COGs) in the set of acI genomes. Both OrthoMCL and BLAST were run using default options [@Fischer2011]. Annotations were assigned to protein clusters by choosing the most common (consensus) annotation among all genes assigned to that cluster. Then, trimmed and merged metatranscriptomic reads from each of the four samples were mapped to a single reference fasta file containing all acI genomes using BBMap (https://sourceforge.net/projects/bbmap/) with the `ambig=random` and `minid=0.95` options. The 95% identity cutoff was chosen as this represents a well-established criteria for identifying microbial species using average nucleotide identity (ANI) [@Konstantinidis2005a], while competitive mapping using pooled acI genomes as the reference ensures that reads map only to a single genome. These results were then used to compute the expression of each COG in each clade.

Next, HTSeq-Count [@Anders2014] was used to count the total number of reads that map to each gene in our acI genome collection. After mapping, the list of counts was filtered to remove those genes that did not recruit at least one read in all four samples. Using the COGs identified by OrthoMCL, the genes that correspond to each COG were then identified.

Within each clade, gene expression for each COG was computed on a Reads Per Kilobase Million (RPKM) basis [@Mortazavi2008], while accounting for different sequencing depths across metatranscriptomes and gene lengths within a COG. RPKM counts were then averaged across the four metatranscriptomes and normalized to the median level of gene expression within that clade.

## Availability of Data and Materials

All genomic and metatranscriptomic sequences are available through IMG and NCBI, respectively. A reproducible version of this manuscript is available at https://github.com/joshamilton/Hamilton_acI_2016 (DOI:######).

# Results

## Genome Statistics and Phylogenetic Affiliation

We have assembled a reference genome collection containing 17 SAGs and 19 MAGs from members of the acI lineage. The SAGs, 11 of which have been previously described [@Garcia2013; @Ghylin2014], were generated from four temperate lakes in the United States and Europe, while the MAGs were generated from two temperate lakes in the United States (15 MAGs, nine of which have been previously described [@Bendall2016]), Spanish and American reservoirs (three MAGs [@Ghai2014; @Tsementzi2014]), and a mixed culture from a European temperate lake [@Garcia2015]. The full list of genomes is given in Table 1.

A phylogenetic tree of these genomes is shown in Figure 1, showing that our genome collection contains genome from all three acI clades and seven tribes. As described previously, we were able to classify to the tribe level SAGs using 16S rRNA genes. MAGs were then classified as belonging to a particular tribe if they formed a monophyletic group with a SAG from that tribe. Of note, three MAGs formed a monophyletic group separate from tribes acI-A and acI-B, a topology shared by 16S rRNA gene trees [@Newton2011a]. These genomes belong to clade acI-C, and represent the first genomes identified as belonging to this clade. Additionally, five MAGs fell into one of the seven tribes defined by our SAGs.

Genome completeness estimates for the new genomes range from 51 to 87% (Table 1), with estimated genome sizes between 1 and 2 Mb. The GC content of these genomes was also low (40 to 50%), and both estimated genome size and GC content are consistent with previously-published acI genomes [@Ghai2012; @Garcia2013; @Ghylin2014; @Garcia2015; @Tsementzi2014; @Bendall2016]. Estimated genome size and GC content of clade acI-C were not statistically different from clades acI-A and acI-B.

## Estimated Completeness of Tribe- and Clade-Level Composite Genomes

Metabolic network reconstructions created from these genomes will likely be missing reactions, as the underlying genomes are incomplete. Previous studies have examined the effect of genome incompleteness on the predicted seed set [@Borenstein2008]. Starting with a reference (complete) network, this study used graph-theoretical analysis (using the formal definition of a seed compound) to show that the percentage of topologically-correct seed compounds (true positives) is approximately equal to the completeness of the reaction network [@Borenstein2008], and the number of false positives is approximately equal to the incompleteness of the network [@Borenstein2008]. As individual acI genomes are incomplete, we constructed composite genomes at higher taxonomic levels (e.g., tribe and clade) to increase genome completeness for more accurate seed identification at that taxonomic level.

Using conserved single-copy marker genes [@Parks2015], we estimated the completeness of tribe- and clade-level composite genomes to determine the finest level of taxonomic resolution at which we could confidently compute seed compounds, using genome completeness as a proxy for metabolic reaction network completeness (Figure 2). With the possible exception of tribe acI-B1, tribe-level composite genomes are estimated to be incomplete (Figure 2A). For tribe acI-B1, the estimated completeness does not appear to increase beyond 94% once the composite genome contains seven SAGs and MAGs. At the clade level, clades acI-A and acI-B are estimated to be complete, as the estimated completeness of both clade-level composite genomes exceeds 99%. (The clade acI-A composite genome reaches this threshold when the composite genome contains nine SAGs and MAGs, while the clade acI-B composite genome reaches this threshold after 11 SAGs and MAGs.) However, the acI-C composite genome remains incomplete (Figure 2B), as it only contains 75% of the marker genes. As a result, seed compounds were calculated for composite clade-level genomes, with the understanding that some true seed compounds for the acI-C clade will not be predicted.

## Making Sense of Seed Compounds via Protein Clustering and Metatranscriptomic Mapping

An organism's seed set contains all of the metabolites which cannot be synthesized by its metabolic network. These metabolites may represent both auxotrophies and nutrient sources. In the latter case, genes associated with the consumption of these compounds should be expressed. However, because seed compounds were computed from each clade's composite metabolic network graph, genes associated with the consumption of seed compounds may be present in multiple genomes within the clade. To facilitate the linkage of metatranscriptome measurements to seed compounds, we decided to map metatranscriptome samples to the "pan-genome" of each clade. To construct the pan-genome, we used OrthoMCL [@Li2003] to identify clusters of orthologous groups (COGs) in the set of acI genomes, and defined the pan-genome of a clade as the union of all COGs present in at least one genome. We then used BBMap to map metatranscriptome reads to our reference genome collection, and counted the unique reads which map to each actinobacterial COG.

OrthoMCL identified a total of 5013 protein clusters across the three clades (Table S3). Of these, 1078 (22%) represent core genes, defined here as being present in at least one genome belonging to each clade. We note that our aim was not to determine the “core genome" in the conventional sense, but to identify gene sets that represent the conserved metabolic capabilities of the acI ineage, with the constraint of incomplete individual genomes. The COGs were unequally distributed across the three clades, with clade acI-A genomes containing 3175 COGs (63%), clade acI-B genomes containing 3459 COGs (69%), and clade acI-C genomes containing 1365 COGs (27%). Of these COGs, 650 were expressed in clade acI-A, 785 in clade acI-B, and 849 in clade acI-C (Table S4). Among expressed genes, the median log2 average RPKM value was 10.3 in clade acI-A, 10.2 in clade acI-B, and 9.0 in clade acI-C.

Sequencing of cDNA from all four samples yielded approximately 160 billion paired-end reads. After merging, filtering, and _in-silico_ rRNA removal, approximately 81 billion, or 51% of the reads remained (Table S1). These reads were subsequently mapped against our acI SAG and MAG collection. We used the metatranscriptomic reads that mapped to each clade as a proxies for relative activity (Table S2).

## Computation and Evaluation of Potential Seed Compounds

Seed compounds were computed for each clade, using the composite metabolic network graph for that clade (Figure 3, and Figures S1 to S3). A total of 125 unique seed compounds were identified across the three clades (Table S5). Additional details are available in the Supplemental Online Material.

Seed compounds were predicted using the results of an automated annotation pipeline, and as such are likely to contain inaccuracies (e.g., due to missing or incorrect annotations). As a result, we screened the set of predicted seed compounds to identify those that represented biologically plausible auxotrophies and nutrients, and manually curated this subset to obtain a final set of auxotrophies and nutrient sources. The Supplemental Online Material contains a series of brief vignettes explaining why select compounds were retained or discarded based on their biological (im)plausibility, and provides examples of manual curation efforts applied to biologically plausible compounds. For a plausible auxotrophy, we screened the genomes for the canonical biosynthetic pathway(s) for that compound, and retained those compounds for which the biosynthetic pathway was incomplete. For a plausible nutrient source, we screened the genomes for the canonical degradation pathway(s) for that compound, and retained those compounds for which the degradation pathway was complete. Tables S6 and S7 contain the final set of proposed auxotrophies and nutrients, respectively, for clades acI-A, acI-B, and acI-C.

## Auxotrophies and Nutrient Sources of the acI Lineage

Seed set analysis yielded seven autotrophies that could be readily mapped to ecophysiological attributes of the acI lineage (Figure 4a). In all three clades, beta-alanine was identified as a seed compound, suggesting an auxotrophy for pantothenic acid (Vitamin B5), a precursor to coenzyme A formed from beta-alanine and pantoate. In bacteria, beta-alanine is typically synthesized via the aspartate decarboxylation, and we were unable to identify a candidate gene for this enzyme (aspartate 1-decarboxylase, E.C. 4.1.1.11) in any acI genome. Pyridoxine 5'-phosphate and 5'-pyridoxamine phosphate (forms of the enzyme cofactor pyridoxal 5'-phosphate, Vitamin B6) were also predicted to be seed compounds, and numerous enzymes in the biosynthesis of these compounds were not found in the genomes.

Clades within the acI lineage also exhibited distinct auxotrophies. Clade acI-A was predicted to be auxotrophic for the cofactor tetrahydrofolate (THF or Vitamin B9), and numerous enzymes for its biosynthesis were missing. This cofactor plays an important role in the metabolism of amino acids and vitamins. In turn, clade acI-B was predicted to be auxotrophic for adenosylcobalamin (Vitamin B12), containing only a single reaction from its biosynthetic pathway. Finally, acI-C was predicted to be auxotrophic for the nucleotide uridine monophosphate (UMP, used as a monomer in RNA synthesis) and the amino acids lysine and homoserine. In all cases multiple enzymes for the biosynthesis of these compounds were not found in the acI-C genomes. However, with the exception of adenosylcobalamin, we did not identify transporters for any of these compounds. Furthermore, because the acI-C composite genome was estimated to be around 75% complete, we cannot rule out the possibility that the missing genes might be found in when additional genomes are recovered.

A number of seed compounds were predicted to be nutrients, compounds which can be degraded by members of the aCI lineage. Both clades acI-A and acI-B were predicted to use D-altronate and trans-4-hydroxy proline as nutrients, and acI-B was additionally predicted to use glycine betaine. These compounds indicate that the acI may participate in the turnover of plant- and animal-derived organic material in freshwater systems: glycine betaine is an important osmolyte in plants [@Ashraf2007], D-altronate is produced during degradation of galacturonate, a component of plant pectin [@Mohnen2008], and trans-4-hydroxy-L-proline is a major component of animal collagen [@Eastoe1955].

Finally, all three clades were predicted to use as nutrients the di-peptides ala-leu and gly-pro-L and the sugar maltose. Clades acI-A and acI-C were also predicted to consume the polysaccharides stachyose, manninotriose, and cellobiose. In all cases, these compounds were associated with reactions catalyzed by peptidases or glycoside hydrolases (Table S8 and S9). We used these annotations to define nutrient sources, rather than using the predicted seed compounds themselves. Figure 4b summarizes predicted nutrients for the acI lineage.

Among these nutrient sources were di- and polypeptides, predicted to be released from both cytosolic- and membrane-bound aminopeptidases. As discussed below, we identified a number of transport proteins capable of transporting these released residues. In Lake Mendota, these aminopeptidases were expressed in clades acI-A and acI-B at around 70% of the median gene expression levels, while they were expressed at up to twice the median in clade acI-C (Table S8). This finding agrees with MAR-FISH and CARD-FISH studies, which confirm the ability of acI bacteria to consume a variety of amino acids [@Salcher2010; @Salcher2013].

All three clades were predicted to encode an alpha-glucosidase, which in Lake Mendota was expressed most strongly in clade acI-C, at approximately 116% of the median (Table S9). Clades acI-A and acI-C also encode a beta-glucosidase, though it was not expressed. Both of these enzymes release glucose monomers, which acI is known to consume [@Buck2009; @Salcher2013]. Furthermore, these two clades encode an alpha-galactosidase and multiple maltodextrin glucosidases (which frees maltose from maltotriose), both of which were only expressed in clade acI-C over our sampling period. The alpha-galactosidase had a log2 average RPKM expression value of 2.5 times the median, while the maltodextrin glucosidases were expressed at approximately 20% of the median (Table S9).

These results suggest the acI lineage is capable of consuming a diverse array of peptides and polysaccharides. We hypothesize that the acI obtain these peptides from the products of cell lysis, and participate in the turnover of high molecular weight dissolved organic compounds, such as starch, glycogen, and cellulose.

## Compounds Transported by the acI Lineage

All acI clades encode for and expressed a diverse array of transporters (Figure 5 and Tables S10 and S11). Consistent with the presence of intra- and extra-cellular peptidases, all clades contain numerous genes for the transport of peptides and amino acids, including multiple oligopeptide and branched-chain amino acid transporters, as well as two distinct transporters for the polyamines spermidine and putrescine. All clades also contain a transporter for ammonium. As averaged over the 24-hour sampling period, the ammonium, branched-chain amino acid, and oligopeptide transporters had expression values above the median, with expression values for the substrate-binding protein ranging from 2 to 325 times the median (Table S10). In contrast, while all clades expressed some genes from the polyamine transporters, only clade acI-B expressed the spermidime/putrescine binding protein, at approximately 75 times the median (Table S10). Additionally, clade acI-A contains a third distinct branched-chain amino acid transporter, composed of COGs not found in clades acI-B or acI-C. This transporter was not as highly expressed as the shared transporters, with the substrate-binding protein not expressed at all (Table S10). Finally, clades acI-A and acI-B also contain a transporter for glycine betaine, which was only expressed in clade acI-A, approximately 35 times the median (Table S10). However, because these observations were made at a single site at a single point in time, we cannot rule out the possibility that the expression of these transporters changes with space and time.

All clades also strongly expressed transporters consistent with the presence of glycoside hydrolases, including transporters for the sugars maltose (a dimer of glucose) and xylose, with expression values for the substrate-binding protein ranging from 3 to 144 times the median (Table S10). Clades acI-A and acI-B also contain four distinct transporters for ribose, although the substrate-binding subunit was not expressed at the time of sampling (Table S10).

The acI lineage also encodes for and expressed a number of transporters that do not have corresponding seed compounds, including a uracil permease, and a xanthine/uracil/thiamine/ascorbate family permease, both of which are expressed at levels ranging from 11 to 127 times the median (Table S10) during the sampling period. Clades acI-A and acI-B also contain a a cytosine/purine/uracil/thiamine/allantoin family permease, though it was only expressed in clade acI-B at the time of sampling (Table S10). All three clades contain and expressed the high-affinity phosphate specific transport system (Pst), with expression values for the substrate-binding protein ranging from 29 to 69 times the median (Table S10). In addition, clade acI-A contains but did not express a transporter for cobalamin (Vitamin B12), and both clades acI-A and acI-B contain but did not express transporters for thiamin (Vitamin B1) and biotin (Vitamin B7) (Table S10). Despite predicted auxotrophies for pantothenic acid and pyridoxal 5'-phosphate, we were unable to find transporters for these two compounds. However, as annotation of transport proteins is an active area of research (see discussion in [@Saier2014]), transporters for these vitamins may yet be present in the genomes.

Finally, all three clades expressed actinorhodopsin, a light-sensitive opsin protein that functions as an outward proton pump [@Sharma2008]. In all clades, actinorhodopsin was among the top seven most highly-expressed genes at the time of sampling (Table S4), with expression values in excess of 300 times the median in all three clades (Table S4). Given that many of the transport proteins are ATP-binding cassette (ABC) transporters, we speculate that actinorhodopsin may facilitate maintenance of the proton gradient necessary for ATP synthesis. Coupled with high expression levels of diverse transporters, this result suggests that acI functioned as photoheterotrophs during our sampling period. However, it remains to be seen if this behavior is a general feature of acI ecology or restricted to the specific conditions of our sampling period.

The presence of multiple branched-chain amino acid and oligopeptide transporters attests to the importance of these compounds to acI's lifestyle. These ABC transporters are composed of four subunits, including two membrane-associated ATPases and two transmembrane proteins that generally determine the substrate specificity of the transporter [@Higgins1992]. We identified a total of ten distinct oligopeptide transporters within our 36 freshwater acI genomes (Table S10), each with a unique transmembrane (oligopeptide-binding) protein. Six of these transporters are found in all three clades, while the remaining four are present in just one or two clades (Table S10). Similarly, we identified a total of six distinct branched-chain amino acid transporters. In these transporters, an amino acid-binding protein, rather than the transmembrane proteins, determines the substrate specificity [@Adams1990]. Five of the six transporters in our genomes contain the same four transmembrane and ATPase subunits, differing only in the amino acid binding subunit (Table S10). Of these five distinct amino acid binding proteins, only one is found in all three clades, with the others being found in just one or two clades (Table S10). The diversity of these transporters both within and between clades suggests the acI are adapted to a variety of amino acids and oligopeptides, with further specialization within each clade.

# Discussion

This study introduces the use of high-throughput metabolic network reconstruction and the seed set framework to predict auxotrophies and nutrient sources of uncultivated microorganisms from incomplete genome sequences. By leveraging multiple genomes from related populations, we were able to construct composite genomes for higher taxonomic levels. Obviously this masks differences among populations and individual cells, and may sometimes overestimate the shared gene content of a clade or group. However, it provides a framework that can be used to generate new hypotheses about the substrates used by members of a defined phylogenetic group, even when only draft genomes are available. As metagenomic assembly and binning techniques and single cell based methods improve and complete genomes become available, we anticipate our approach being applied to individual microbial genomes.

Our predictions of substrate use capabilities of the acI lineage are largely congruent with previous genome-based studies, based on smaller but manually curated genome collections, indicating that the use of automatic metabolic network reconstructions yields similar predictions to manual metabolic reconstruction efforts. In particular, this study predicts that the consumption of N-rich compounds is a universal feature of the acI lineage, with all three clades predicted to consume ammonium, branched-chain amino acids, the polyamines spermidine and putrescine, and di- and oligopeptides. We provide new evidence for further specialization within each clade, identifying unique substate binding proteins for some of their amino acid and peptide transporters. Furthermore, we confirm the ability of all three clades to consume xylose and maltose, and of clades acI-A and acI-B to consume ribose. However, the possibility that acI-C consumes ribose cannot yet be ruled out, because our acI-C composite metabolic network graph remains incomplete. Our analysis also made novel predictions, including the presence of beta-glucosidases, as well as alpha- and beta-galactosidases, in clades acI-A and acI-C.

Our analysis also suggests that auxotrophies for some vitamins may be universal features of the lineage, as we predict all clades to be auxotrophic for pantothenic acid and pyridoxal 5'-phosphate (Vitamins B5 and B6). However, our analysis does not identify riboflavin (Vitamin B2), niacin (Vitamin B3), or folic acid (Vitamin B9) as auxotrophies for clade acI-B, a result that had been previously published for a member of the acI-B2 tribe [@Garcia2015]. This difference may arise because we are analyzing the metabolism of the entire clade, while previous predictions were made on the basis of a single genome [@Garcia2015]. We also predict new auxotrophies within the acI lineage, including THF (clade acI-A), and lysine, homoserine, and UMP (clade acI-C). These results provide additional support to the hypothesis that distributed metabolic pathways and metabolic complementarity may be common features of freshwater bacterial communities [@Garcia2015; @Garcia2016].

This study also presents the first combined genomic and metatranscriptomic analysis of a freshwater microbial lineage. Transport proteins were among the most highly expressed in the acI genomes, and the expression of multiple amino acid transporters may facilitate uptake of these labile compounds. We also observed differences in the relative expression of these transporters, which may point to differences in acI's affinity for these substrates. The actinorhodopsin protein was highly expressed, and may facilitate synthesis of the ATP needed to drive acI's many ABC-type transporters.

A close comparison of our predictions to previous studies of the acI lineage reveals some important limitations of the seed set framework and automatic metabolic reconstructions. First, the seed set framework only identifies compounds that the metabolic network __must__ obtain from its environment, and will fail to identify compounds that the organism can acquire from its environment but can also synthesize. For example, members of clades acI-A and acI-B are capable of consuming branched-chain amino acids [@Ghylin2014; @Garcia2015], but can also synthesize them. Thus, these compounds were not identified as seed compounds. However, transport reactions for branched-chain amino acids were identified, and applying the seed set framework to a reconstruction that includes transport reactions would identify extracellular forms of the branched-chain amino acids as seed compounds.

Second, automatic metabolic network reconstructions may not fully capture an organism's metabolic network (e.g., due to missing or incorrect genome annotations). For example, previous genome-based studies have suggested acI harbor cyanophycinase and chitinase, enzymes that allow them to breakdown the cyanobacterial peptide cyanophycin and NAG, respectively [@Garcia2013]. Manual inspection revealed that KBase annotated these putative enzymes as hypothetical proteins, and we could not identify transporters for these compounds in the metabolic network reconstruction. As genome and protein annotation are active areas of research, we anticipate that advances in these areas will continue to improve the accuracy of automatic metabolic network reconstructions.

# Conclusions

In this study, we examined the ecological niche of uncultivated acI bacteria using automatic metabolic network reconstructions and the seed set framework. Predicted seed compounds include peptides and saccharides, many of which acI have been observed to consume _in situ_, as well as newly predicted auxotrophies for vitamins and amino acids. Many predictions were corroborated by a metatranscriptome analysis in a lake with abundant acI members. Our high-throughput approach easily scales to 100s and 1000s of genomes, and enables a focused metabolic analysis by identifying those compounds through which an organism interacts with its environment. Finally, the seed set framework enables additional reverse ecological analyses, which promise to predict the interactions between microbial species in complex environments [@Levy2012].

# Acknowledgements

We thank past members of the McMahon lab for collecting water samples for single-cell sequencing and metagenomic sequencing. We thank XXX, YYY, and ZZZ at the US Department of Energy Joint Genome Institute (JGI) for their assistance with data analysis. This work was supported through the JGI Community Science Program. The work conducted by the JGI, a DOE Office of Science User Facility, is supported by the Office of Science of the U.S. Department of Energy under Contract No. DE-AC02-05CH11231. This material is based upon work that is supported by the National Institute of Food and Agriculture, U.S. Department of Agriculture, under award number 2016-67012-24709 to JJH and WIS01789 to KDM. KDM acknowledges funding from the United States National Science Foundation (NSF) Microbial Observatories program (MCB-0702395), the NSF Long Term Ecological Research program (NTL-LTER DEB-1440297), an NSF INSPIRE award (DEB-1344254), and a National Oceanic and Atmospheric Administration NOAA Grant #NA10OAR4170070, Wisconsin Sea Grant College Program Project #HCE-25, through NOAA’S National Sea Grant College Program, U.S. Deptartment of Commerce. KBM was also supported in part by the University of Wisconsin System.

# Conflict of Interest

The authors declare no conflict of interest.

# References
